game/Doom/Game/p_setup.cpp

```diff
@@ -1646,6 +1648,36 @@ void P_SetupLevel(const int32_t mapNum, [[maybe_unused]] const skill_t skill) no
         ScriptingEngine::init();                        // PsyDoom: initialize the scripting engine if the map has Lua scripted actions
         MapHash::finalize();                            // PsyDoom: compute the final map hash
         MapPatcher::applyPatches();                     // PsyDoom: apply any patches to original map data that are relevant at this point, once all things have been loaded
+        // PsyDoom: if playing deathmatch or 'no monsters' setting is set, activate all special tagged boss sectors
+        if (gNetGame == gt_deathmatch || Game::gSettings.bNoMonsters) {
+            for (int32_t i = 666; i <= 672; ++i) {
+                line_t dummyLine = {};
+                dummyLine.tag = i;
+                switch (dummyLine.tag) {
+                case 666:
+                    EV_DoFloor(dummyLine, lowerFloorToLowest);
+                    break;
+                case 667:
+                    EV_DoFloor(dummyLine, raiseFloor24);
+                    break;
+                case 668:
+                    EV_DoFloor(dummyLine, lowerFloorToLowest);
+                    break;
+                case 669:
+                    EV_DoFloor(dummyLine, lowerFloorToLowest);
+                    break;
+                case 670:
+                    EV_DoDoor(dummyLine, Open);
+                    break;
+                case 671:
+                    EV_DoFloor(dummyLine, lowerFloorToLowest);
+                    break;
+                case 672:
+                    EV_DoDoor(dummyLine, Open);
+                    break;
+                }
+            }
+        }
     #else
```

Comment:
What if we would limit this to deathmatch only and make it an optional toggle? The original reason for adding this in is because some of the original maps have areas that are blocked off and are not accessible in deathmatch.

A few examples of this are:
- Phobos Anomaly: Difficult to navigate map because the outside is segregated from the inside. You can go from the outside to the inside via teleports, but not vice-versa, unless you die and respawn outside.
- Tower of Babel: You can't exit the map
- Redemption Denied: Can't exit
- The Marshes: Can't exit

Reply:
I made the changes here:
https://github.com/ts-miller/PsyDoom/commit/bef02ba7e362ddf3fb64416ca292728bce7d8b14